### DISCUSSION TITLE
HM-MOD-RPI and ZigBee-Stick

### DISCUSSION THREAD

**Discussion by ThomasRi333** [OpenCCU/OpenCCU#2182] [🇺🇸 Q&A] [answered]
### Describe the issue you are experiencing

When activating the GPIO-Ports for the modul, Zigbee (at least Conbee 2) stops working

### Describe the behavior you expected

When activating GPIO-Ports, no influence on USB should happen

### Steps to reproduce the issue

1. Clean Install (New SD, newest Home-Asistant image)
2.Starting Zigbee (ZHA) and configuring 1 or 2 devices
3. Restart Homeassistant, no problems
4. Editing Config.TXT to activate HM-MOD-RPI
5. Starting new, Raspberrymatic works perfectly, Zigbee does not work (No Response, no pairing possible)
...

### What is the version this bug report is based on?

3.65.6.20220723

### Which base platform are you running?

rpi4 (RaspberryPi4)

### Which HomeMatic/homematicIP radio module are you using?

RPI-RF-MOD

### Anything in the logs that might be useful for us?

```txt
2023-01-21 17:21:53.445 ERROR (MainThread) [zigpy.zcl] [0x4DBF:2:0x0406] Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/zigpy/zcl/__init__.py", line 358, in reply
    return await self._endpoint.reply(
  File "/usr/local/lib/python3.10/site-packages/zigpy/endpoint.py", line 256, in reply
    return await self.device.reply(
  File "/usr/local/lib/python3.10/site-packages/zigpy/device.py", line 384, in reply
    return await self.request(
  File "/usr/local/lib/python3.10/site-packages/zigpy/device.py", line 297, in request
    await self._application.request(
  File "/usr/local/lib/python3.10/site-packages/zigpy/application.py", line 690, in request
    await self.send_packet(
  File "/usr/local/lib/python3.10/site-packages/zigpy_deconz/zigbee/application.py", line 402, in send_packet
    await self._api.aps_data_request(
AttributeError: 'NoneType' object has no attribute 'aps_data_request'
```

### Additional information

Sorry Jens, i dont know, if this is your part or a problem of Conbee.
I tried to migrate to only one Raspy.

But : Your work on Raspberrymatic is absolute great !

**Comment by jens-maus** [comment]
Converted this issue to a discussion because this can't be a technical issue that we (RaspberryMatic) can solve or is solvable.

To me, this sounds like there are potential RF interferences when using both RF modules (Homematic+Zigbee) in its near vicinity. Thus, use a 2-3 m long USB extender cable to make sure the Zigbee stick is far away from the RaspberryPi. That's the only thing I could think of that might be the reason for it.

**Reply by ThomasRi333** [reply]
Hi Jens, 
both Pis are mounted on a shelf, without any distance between. So long, i got not problems.
I tried a extension cable, but no luck at all...

**Reply by Baxxy13** [reply]
Is CUxD installed inside the RM-AddOn?

**Comment by mash727** [comment]
Hi I had the same problem untill I read the docs and forum on CUXD. After I
deactivated  the tty interface all worked perfectly. I now have a pi4 with
a HMiP plugged directly in USB3 and a Sonoff dongel in USB 2 around half a
meter away. The whole system operates perfectly in an intensive WiFi
environment. Not sure if this helps you.

**Reply by ThomasRi333** [reply]
Thanks for the info, yes I use CUXD, and I haven’t read the docs .
I will give it a try … perhaps tomorrow.

**Reply by ThomasRi333** [reply]
It works... i cant believe it.
(Blame on me for not reading the 196 pages of Cuxd-manual, but it worked until now really perfect)
Due to a missing USB extension, Conbee is plugged in directly in the Pi, but i get no errors at all !!!
Thanks all for the great help

**Comment by akula3** [comment, marked as answer]
I had also a problem with running RPI-RF-MOD, Zigbee Sonoff dongle and HomeAssistant on RPi4.
Extended USB cable didn't help.
Thanks to hint from @mash727 , it works now for half an hour ! The distance between RPI-RF-MOD and Zigbee Sonoff dongle is about 40cm. Zigbee Sonoff dongle is only with a short USB cable extension connected.
Just add in the config file from CUXD an additional line with `TTYPARAM=NONE`

**Comment by Rogash1** [comment]
Hi,
I'm currently trying to set up RaspberryMatic and failing to set up a HM-MOD-RPI-PCB in parallel with a Phoscon RaspBee II. Since both modules would occupy the same pins on the Raspberry Pi, I have already enabled Uart 1 in the config.txt of my Raspberry Pi 4 and temporarily connected the HM-MOD-RPI-PCB with Dupont cables to the corresponding pins.
I'm using RaspberryMatic for the first time and can't find an option to set which UART and other pins the addon should use to communicate with the HM-MOD-RPI-PCB module.

Does anyone have a hint on how to configure the pins used so RaspberryMatic can recognize the HM-MOD-RPI-PCB module?

**Reply by jens-maus** [reply]
Sorry, but this is not possible and there is no way to change the used uart for the homematic module. So if you have two HATs requiring the same UART pins you will have to move the other module to different pins and refrain from trying to use a different uart for the homematic rf HAT. This is also röthe case because the homematic module requires a special uart driver to be used (generic_raw_uart) and will bot work nicely with a standard tty serial driver.
